File: app/src/main/java/com/romainpiel/darktown/HeadingSpan.java
package com.romainpiel.darktown;

import android.text.TextPaint;
import android.text.style.MetricAffectingSpan;

public class HeadingSpan extends MetricAffectingSpan implements MarkDownSpan {

    private static final float RELATIVE_SIZE = 1.2f;

    @Override
    public void updateDrawState(TextPaint tp) {
        tp.setFakeBoldText(true);
        tp.setTextSize(tp.getTextSize() * RELATIVE_SIZE);
    }

    @Override
    public void updateMeasureState(TextPaint tp) {
        tp.setFakeBoldText(true);
        tp.setTextSize(tp.getTextSize() * RELATIVE_SIZE);
    }
}

File: app/src/main/kotlin/com/romainpiel/darktown/HeadingSpan.kt
package com.romainpiel.darktown

import android.text.TextPaint
import android.text.style.MetricAffectingSpan

class HeadingSpan : MetricAffectingSpan(), MarkDownSpan {

    override fun updateDrawState(tp: TextPaint) {
        tp.isFakeBoldText = true
        tp.textSize = tp.textSize * RELATIVE_SIZE
    }

    override fun updateMeasureState(tp: TextPaint) {
        tp.isFakeBoldText = true
        tp.textSize = tp.textSize * RELATIVE_SIZE
    }

    companion object {

        private val RELATIVE_SIZE = 1.2f
    }
}
